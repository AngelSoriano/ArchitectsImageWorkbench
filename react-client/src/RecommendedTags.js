var React = require('react');
import {Button, ButtonToolbar} from 'react-bootstrap';


var RecommendedTags = React.createClass({
    getDefaultProps: function() {
        // At some point we would have a backend route to get the recommended tags and store them in array
        return {text: ["tag1", "tag2", "tag3", "tag4", "tag5", "tag6"]}
    },
    render: function() {

        return (
            <div>
                <ButtonToolbar>
                    <Button bsStyle="primary">{this.props.text[0]}</Button>
                    <Button bsStyle="success">{this.props.text[1]}</Button>
                    <Button bsStyle="info">{this.props.text[2]}</Button>
                    <Button bsStyle="warning">{this.props.text[3]}</Button>
                    <Button bsStyle="danger">{this.props.text[4]}</Button>
                    <Button bsStyle="default">{this.props.text[5]}</Button>
                </ButtonToolbar>
            </div>
        )
    }
});


module.exports = RecommendedTags;