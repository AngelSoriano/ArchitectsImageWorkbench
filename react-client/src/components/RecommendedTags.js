import React, {Component} from 'react'
import {Button, ButtonToolbar} from 'react-bootstrap';
import TagService from '../service/TagService'

class RecommendedTags extends Component {
    render() {
        return (
            <div className="RecommendedTags">
                <ButtonToolbar className="TagsBar">
                    {this.renderTags()}
                </ButtonToolbar>
            </div>
        )
    }

    constructor(props) {
        super(props);

        this.state = {
            tags: ['architecture', 'building', 'modern', 'office', 'chair', 'indoors']
        };

        this.getTagList = this.getTagList.bind(this);
        this.getImageIds = this.getImageIds.bind(this);
        this.getTagList()
    }

    getImageIds() {
        var imageIdsAndConfidenceLevels = JSON.parse(this.props.imageList)

        var imageIds = [];
        for (var imageId in imageIdsAndConfidenceLevels) {
            imageIds.push(imageId)
        }

        console.log(imageIds)
        return imageIds
    }

    getTagList() {
        TagService.retrieve(this.getImageIds(), (tags) => {
            this.setState({tags: JSON.parse(tags)})
        })
    }

    renderTags() {
        return this.state.tags.map(name => (
            <Button className="TagsBarItem"
                    key={name}
                    name={name}
            >{name}</Button>
        ))
    }

}

export default RecommendedTags