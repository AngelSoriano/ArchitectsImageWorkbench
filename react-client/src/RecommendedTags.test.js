/**
 * Created by anaperez on 3/8/17.
 */
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import RecommendedTags from './RecommendedTags';

describe(RecommendedTags, () => {
    //tests go here
    it('renders and matches our snapshot', () => {
        const component = renderer.create(
            <RecommendedTags/>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});
