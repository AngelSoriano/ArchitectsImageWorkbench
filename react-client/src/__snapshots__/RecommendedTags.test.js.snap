exports[`function (props, context, updater) {
      // This constructor gets overridden by mocks. The argument is used
      // by mocks to assert on what gets mounted.

      if (process.env.NODE_ENV !== 'production') {
        process.env.NODE_ENV !== 'production' ? warning(this instanceof Constructor, 'Something is calling a React component directly. Use a factory or ' + 'JSX instead. See: https://fb.me/react-legacyfactory') : void 0;
      }

      // Wire up auto-binding
      if (this.__reactAutoBindPairs.length) {
        bindAutoBindMethods(this);
      }

      this.props = props;
      this.context = context;
      this.refs = emptyObject;
      this.updater = updater || ReactNoopUpdateQueue;

      this.state = null;

      // ReactClasses doesn't have constructors. Instead, they use the
      // getInitialState and componentWillMount methods for initialization.

      var initialState = this.getInitialState ? this.getInitialState() : null;
      if (process.env.NODE_ENV !== 'production') {
        // We allow auto-mocks to proceed as if they're returning null.
        if (initialState === undefined && this.getInitialState._isMockFunction) {
          // This is probably bad practice. Consider warning here and
          // deprecating this convenience.
          initialState = null;
        }
      }
      !(typeof initialState === 'object' && !Array.isArray(initialState)) ? process.env.NODE_ENV !== 'production' ? invariant(false, '%s.getInitialState(): must return an object or null', Constructor.displayName || 'ReactCompositeComponent') : _prodInvariant('82', Constructor.displayName || 'ReactCompositeComponent') : void 0;

      this.state = initialState;
    } renders and matches our snapshot 1`] = `
<div>
  <div
    className="tagsBar btn-toolbar"
    role="toolbar">
    <button
      className="btn btn-primary"
      disabled={false}
      type="button">
      tag1
    </button>
    <button
      className="btn btn-success"
      disabled={false}
      type="button">
      tag2
    </button>
    <button
      className="btn btn-info"
      disabled={false}
      type="button">
      tag3
    </button>
    <button
      className="btn btn-warning"
      disabled={false}
      type="button">
      tag4
    </button>
    <button
      className="btn btn-danger"
      disabled={false}
      type="button">
      tag5
    </button>
    <button
      className="btn btn-default"
      disabled={false}
      type="button">
      tag6
    </button>
  </div>
</div>
`;
